src/main/scala/com/gu/anghammarad/models/Serialization.scala

```diff
@@ -1,38 +1,74 @@
 package com.gu.anghammarad.models
 
+import com.gu.anghammarad.Enrichments._
 import io.circe.Decoder.Result
 import io.circe._
-import io.circe.generic.semiauto._
 import io.circe.parser._
-import io.circe.syntax._
 
 import scala.util.{Failure, Success, Try}
-import com.gu.anghammarad.Enrichments._
 
 
 object Serialization {
-  def parseMapping(jsonStr: String): Try[Mapping] = {
-    val tryJson = parse(jsonStr).fold(
+
+  def parseAllMappings(jsonStr: String): Try[List[Mapping]] = {
+    val allMappings = for {
+      json <- parse(jsonStr)
+      rawMappings <- json.hcursor.downField("mappings").as[List[Json]]
+      mappings <- rawMappings.traverseE(parseMapping)
+    } yield mappings
+
+    allMappings.fold(
       err => Failure(err),
-      json => Success(json)
+      suc => Success(suc)
     )
+  }
 
+  private[models] def parseMapping(json: Json): Result[Mapping] = {
+    val hCursor = json.hcursor
+    for {
+      rawTargets <- hCursor.downField("target").as[Json]
+      rawContacts <- hCursor.downField("contacts").as[Json]
+      targets = parseTargets(rawTargets)
+      contacts = parseContacts(rawContacts)
+    } yield Mapping(targets, contacts)
+  }
 
+  private[models] def parseTargets(jsonTargets: Json): List[Target] = {
+    def parseTarget(key: String, value: String): Option[Target] = {
+      key.toLowerCase match {
+        case "stack" => Some(Stack(value))
+        case "stage" => Some(Stage(value))
+        case "app" => Some(App(value))
+        case "awsaccount" => Some(AwsAccount(value))
+        case _ => None
+      }
+    }
 
-    tryJson.map { json =>
-      val mappings = json.hcursor.downField("mappings").as[List[Json]]
+    val c: HCursor = jsonTargets.hcursor
+    // TODO: do we want to return an empty list, or throw a error here?
```

Comment:
I would suggest throwing an error in the future - if we don't have a target, we can't notify - and that's the primary function of this service, so we might as well fail fast!